docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
I thing exchange/agree is an interface that makes sense
The first sets up the exchange properties, the second gives you the shared_key.
You may be setting up one exchange() and then call agree() many times over, each time with a different party.